plugins/grafana-custom-plugins/grafana-chord-plugin/src/ChordPanel.tsx

```diff
@@ -8,14 +8,17 @@ interface Props extends Themeable2, PanelProps<ChordOptions> {}
 
 export const UnthemedChordPanel: React.FC<Props> = ({ options, data, width, height, theme }) => {
   const d3Container = useRef(null);
-
   // GetFieldVal reads field's value from data series
   function GetFieldVal(fieldName: string) {
     return data.series
       .map((series: DataFrame) => series.fields.find((field: any) => field.name.toLowerCase() === fieldName.toLowerCase()))
       .map((field: any) => {
         let record = field?.values as any;
-        return record?.buffer;
+        if(Array.isArray(record)) {
```

Comment:
Out of curiosity, can we just do `let record = field.display!(field.values)`? I'm not sure whether it will run into issues, just referencing https://github.com/grafana/grafana/blob/16d0aff26707b80fda408708d1c67bb2fe0586e9/packages/grafana-ui/src/components/Table/utils.ts#L521
The `else` block is a bit confusing for me. Does it mean if the record is not an array, then we should just return null?

Reply:
The reason I check if record is array is that I found that record is different type when I use jest and webpack.
For webpack, it is a vectorArray and record?.buffer is an array. But for jest, record is array already and record?.buffer will be "undefined", which will break the test.

Reply:
could you help try if `let record = field.display!(field.values)` will work for both jest and webpack?

If not, could you add a comment in the code for the above explanation?

Reply:
That doesn't work for jest.
```
field.display is not a function
TypeError: field.display is not a function
```
I've added comments to explain the reason. Please take a look to see if it is clear enough?